pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -274,13 +288,14 @@ func (rc *reconciler) reconcile() {
 			}
 		}
 	}
+}
 
-	// Ensure devices that should be detached/unmounted are detached/unmounted.
+func (rc *reconciler) unmountDetachDevices() {
 	for _, attachedVolume := range rc.actualStateOfWorld.GetUnmountedVolumes() {
 		// Check IsOperationPending to avoid marking a volume as detached if it's in the process of mounting.
 		if !rc.desiredStateOfWorld.VolumeExists(attachedVolume.VolumeName) &&
 			!rc.operationExecutor.IsOperationPending(attachedVolume.VolumeName, nestedpendingoperations.EmptyUniquePodName) {
-			if attachedVolume.GloballyMounted {
+			if attachedVolume.DeviceMayBeMounted() {
```

Comment:
not a big issue, but this function could be implemented locally?

Reply:
I am thinking of leaving it as it is - since the resulting if statement looks quite big when placed inline and I do not like making a local helper function in this file.. :(

Reply:
I mean you can have a local function to check device maybe mounted?